### ISSUE TITLE
Twitch Login Strategy Error :: Parameter redirect_uri does not match registered URI

### ISSUE THREAD

**Issue by b3ck** [Requarks/wiki#1268]
### Actual behavior

After adding in all of the correct information (from Twitch Dev Console) in the Twitch Strategy Module I tried to login and get the following error: 
```Parameter redirect_uri does not match registered URI```

![Wiki.js Twitch Strategy Error](https://i.imgur.com/Osh4duR.png "Wiki.js Twitch Strategy Error")

---
### Expected behavior

Log into Wiki.js with Twitch Username.

---
### Steps to reproduce the behavior

Try to log in?

---
### System Information:
Wiki.js: ```Stable 2.0.12```
OS: ```Ubuntu 16.04 x86_64```
DB: ```PostgreSQL 9.5```
SSL Enabled: ```Let's Encrypt```

**Comment by nicolasg-playster** [comment]
Make sure the redirect URI you entered in Twitch matches the one shown in Wiki.js in Admin > Authentication > Twitch in the reference panel.

**Comment by NGPixel** [comment]
Make sure the redirect URI you entered in Twitch matches the one shown in Wiki.js in Admin > Authentication > Twitch in the reference panel.

**Comment by b3ck** [comment]
It does match, I was looking at the code, for the Twitch strategy and it doesn't seem it's using the authorization URL or the scope like in the Discord strategy.. but then again.. I'm not a coder =/

**Comment by b3ck** [comment]
So I went into the ```passport-twitch``` strategy, and replaced the old authorization URLs with the new twitch api, now I need to figure out a way to fetch the user profile as that is my current error:

![Twitch Fetch Error](https://i.imgur.com/XBtYeHk.png "Twitch Fetch Error")

Obviously this is going to happen since I totally switched to a different API, so more investigating is in order, I'll update you when I can.

**Comment by b3ck** [comment]
Yeah.. I don't know what Twitch API you guys have this based on, there is v3, v5 and now the "New Twitch API" so.. I don't know.. getting kinda lost in all of these APIs so I'm going to stop for now, let me know if you come up with anything, thanks 😄👍

**Comment by b3ck** [comment]
So I changed everything back to default and replaced ```line 16``` in the ```/server/modules/authentication/twitch/authentication.js``` file;

from: ```callbackURL: conf.callbackURL```
to: ```callbackURL: "https://wiki.b3ck.com/login/twitch/callback"```

And I am no longer getting the: ```Parameter redirect_uri does not match registered URI``` error.

But now I am getting the: ```failed to fetch user profile``` error.

Thoughts?

**Comment by b3ck** [comment]
I found this, it might help : [passport-twitch-new](https://github.com/Nigh7Sh4de/passport-twitch-new)

@NGPixel would that be useful?

**Comment by b3ck** [comment]
This shouldn't be marked as help, this is a **bug**.  The Twitch Login Strategy is broken and needs to be fixed. Thanks.

**Comment by NGPixel** [comment]
The help label is added automatically based on the issue type.

**Comment by b3ck** [comment]
Ahh... Copy that so.. any progress on this issue?

**Comment by b3ck** [comment]
Ahh... Copy that so.. any progress on this issue?

**Comment by b3ck** [comment]
Hello?

**Comment by AvianAeternum** [comment]
Latest is also experiencing this issue with discord oauth

**Comment by TimoKruth** [comment]
Verified against current v2 `main` (`6f042e97`, v2.5.314): this appears fixed.

Current v2 uses `passport-twitch-strategy` 2.2.0 and constructs the callback from the configured site host as `<host>/login/<strategy>/callback`. With a test host of `http://127.0.0.1:3003`, I instantiated the current strategy and inspected its generated Twitch authorization URL; it contained the exact encoded `redirect_uri=http://127.0.0.1:3003/login/twitch/callback`.

This matches the URI an administrator must register in Twitch. I cannot complete a real Twitch login without app credentials, but the redirect URI mismatch in the original v2.0.12 implementation is no longer present.
